internal/storage/localdir_test.go

```diff
@@ -0,0 +1,148 @@
+		testFS = generateFS()
+	})
+	When("a bundle is not stored", func() {
+		Describe("Store", func() {
+			It("should store a bundle FS", func() {
+				Expect(store.Store(ctx, owner, testFS)).To(Succeed())
+				_, err := os.Stat(filepath.Join(store.RootDirectory, fmt.Sprintf("%s.tgz", owner.GetUID())))
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		Describe("Load", func() {
+			It("should fail due to file not existing", func() {
+				_, err := store.Load(ctx, owner)
+				Expect(err).To(WithTransform(func(err error) bool { return errors.Is(err, os.ErrNotExist) }, BeTrue()))
+			})
+		})
+
+		Describe("Delete", func() {
+			It("should succeed despite file not existing", func() {
+				Expect(store.Delete(ctx, owner)).To(Succeed())
+			})
+		})
+	})
+	When("a bundle is stored", func() {
+		BeforeEach(func() {
+			Expect(store.Store(ctx, owner, testFS)).To(Succeed())
+		})
+		Describe("Store", func() {
+			It("should re-store a bundle FS", func() {
+				Expect(store.Store(ctx, owner, testFS)).To(Succeed())
+			})
+		})
+
+		Describe("Load", func() {
+			It("should load the bundle", func() {
+				loadedTestFS, err := store.Load(ctx, owner)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(fsEqual(testFS, loadedTestFS)).To(BeTrue())
+			})
+		})
+
+		Describe("Delete", func() {
+			It("should delete the bundle", func() {
+				Expect(store.Delete(ctx, owner)).To(Succeed())
+				_, err := os.Stat(filepath.Join(store.RootDirectory, fmt.Sprintf("%s.tgz", owner.GetUID())))
+				Expect(err).To(WithTransform(func(err error) bool { return errors.Is(err, os.ErrNotExist) }, BeTrue()))
+			})
+		})
+	})
+})
+
+func generateFS() fs.FS {
+	gen := fstest.MapFS{}
+
+	numFiles := rand.IntnRange(10, 20)
+	for i := 0; i < numFiles; i++ {
+		pathLength := rand.IntnRange(30, 60)
+		filePath := ""
+		for j := 0; j < pathLength; j += rand.IntnRange(5, 10) {
+			filePath = filepath.Join(filePath, rand.String(rand.IntnRange(5, 10)))
+		}
+		gen[filePath] = &fstest.MapFile{
+			Data:    []byte(rand.String(rand.IntnRange(1, 400))),
+			Mode:    fs.FileMode(rand.IntnRange(0600, 0777)),
+			ModTime: time.Now().Add(time.Duration(-rand.IntnRange(0, 10000000)) * time.Second),
+		}
+	}
+	return &gen
+}
+
+func fsEqual(a, b fs.FS) (bool, error) {
+	aMap := fstest.MapFS{}
+	bMap := fstest.MapFS{}
+
+	walkFunc := func(f fs.FS, m fstest.MapFS) fs.WalkDirFunc {
+		return func(path string, d fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+			if d.IsDir() {
+				return nil
+			}
+			data, err := fs.ReadFile(f, path)
+			if err != nil {
+				return err
+			}
+			info, err := d.Info()
+			if err != nil {
+				return err
+			}
+			m[path] = &fstest.MapFile{
+				Data:    data,
+				Mode:    d.Type(),
+				ModTime: info.ModTime(),
+				Sys:     info.Sys(),
+			}
+			return nil
+		}
+	}
+	if err := fs.WalkDir(a, ".", walkFunc(a, aMap)); err != nil {
+		return false, err
+	}
+	if err := fs.WalkDir(a, ".", walkFunc(b, bMap)); err != nil {
```

Comment:
Is `fs.WalkDir(a,` correct?

Reply:
Nope. Copy/paste error. Thanks for catching that!